sdk/index.ts

```diff
@@ -172,14 +183,26 @@ export class HttpRequestBuilder {
     }
     return queryString;
   }
+
+  private renderFormData(): string {
+    let formData = "";
+    const keys = this._formData.keys();
+    const values = this._formData.values();
+    for (let i = 0; i < keys.length; i++) {
+      const value = encodeURIComponent(values[i]);
+      formData += `${keys[i]}=${value}&`;
+    }
+
+    return formData;
+  }
```

Comment:
We need some way to distinguish between urlencoded and multipart/form-data.
For now, just urlencoded is fine, but it would be good to make it clear somehow that it defaults to urlencoded unless a setting is set to change it to multipart/form-data.

Maybe we make a different class to hold `private _formData: Map<string, string> = new Map();`, and set this in the HTTP method. For example:


```ts
export enum HttpFormEncoding {
  UrlEncoded,
  Multipart
}

export class HttpForm {
  private _formData: Map<string, string> = new Map();
  public encoding: HttpFormEncoding;
 
  public constructor(encoding: HttpFormEncoding) { ... }
  
  public setField(...) {}
}
```

then on the HTTP request builder we can call `setForm(httpFormInstance)`, and the HttpRequestBuilder handles serializing it according to the form encoding.

The enum can only have UrlEncoded for now, we can implement multipart later, but at least the API is forward compatible that way.

Reply:
Good thinking, addressed